test/e2e/websocket_test.go

```diff
@@ -147,3 +149,37 @@ func TestWebSocket(t *testing.T) {
 		t.Error(err)
 	}
 }
+
+// TestWebSocketFromZero (1) creates a service based on the `wsserver` image,
+// (2) waits for the service to be scaled down to zero replicas
+// (3) connects to the service using websocket, (4) sends a message, and
+// (5) verifies that we receive back the same message.
+func TestWebSocketFromZero(t *testing.T) {
+	clients := Setup(t)
+
+	names := test.ResourceNames{
+		Service: test.ObjectNameForTest(t),
+		Image:   wsServerTestImageName,
+	}
+
+	// Clean up in both abnormal and normal exits.
+	defer test.TearDown(clients, names)
+	test.CleanupOnInterrupt(func() { test.TearDown(clients, names) })
+
+	// Setup a WebSocket server.
```

Comment:
Eh, this comment is confusing: nothing in the next line says websocket (it is the image above). So I suggest just remove  the comment.

Reply:
I agree that the comment isn't great, but does the absence of mention of 'websocket' in the next line make a comment more valuable here? (though maybe the error case right below provides the necessary context...)